### PR TITLE
Sockets - Carla

### PR DESCRIPTION
**Pull request by carlabosco**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | Control flow with while and if/else statements work mostly the same way.
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | No, online searching worked the same.
What was a challenge you were able to overcome on this assignment? | My usesAvailableLetters in ruby was a lot simpler than the one I ended up with in javascript. I spent a long time trying to simply 'translate' one into the other, but I don't have the same tools in JS.
What has been interesting and positive about learning a new programming language? | Like learning natural languages, I can identify patterns that are helping me learn JS faster than I learned ruby.
What is something to focus on your learnings in JavaScript in the next week? | Practice, practice, practice.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,122 @@
...
+
+const letters = ['A','B','C','D','E','F','G','H',
+'I','J','K','L','M','N','O','P','Q','R','S',
+'T','U','V','W','X','Y','Z'];
+
+const weights = [9, 2, 2, 4, 12, 2, 3, 2, 9, 1, 1, 4,
+2, 6, 8, 2, 1, 6, 4, 6, 4, 2, 2, 1, 2, 1];
+
+const randomLetter = () => {
+  let totalWeigth = 0;
```
Comment (dHelmgren) [inline src/adagrams.js:12]: This logic for drawing a letter works, but it has a problem: you put the letter back in the bag each time, so to speak.

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,122 @@
...
+        return true;
+      } 
+    }
   },
+
+  scoreWord(word) {
+    let points = 0;
+    word = word.toUpperCase();
+
+    for (let i = 0; i < word.length; i +=1) {
```
Comment (dHelmgren) [inline src/adagrams.js:62]: Nice case statement. Could you have gained anything by storing the letters in a js Object?

**Comment by dHelmgren** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | Yes
Small commits with meaningful commit messages | Commit more! this is a new language!
**Code Requirements** |
`drawLetters` method | good
Uses appropriate data structure to store the letter distribution | Yes, but see comment
All tests for `drawLetters` pass | yes
`usesAvailableLetters` method | yes
All tests for `usesAvailableLetters` pass | yes
`scoreWord` method | yes
Uses appropriate data structure to store the letter scores | take another look!
All tests for `scoreWord` pass | yes
`highestScoreFrom` method | yes
Appropriately handles edge cases for tie-breaking logic | yes
All tests for `highestScoreFrom` pass | yes
**Overall** | Great work on this project! In this project you’ve taken some interesting logic and worked it into JavaScript syntax. I’m adding a few comments on places that this could be better and minor suggestions for improvement, but overall, you’ve done a great job of practicing syntax.
